guides/release/components/actions-and-events.md

```diff
@@ -0,0 +1,609 @@
+  @onConfirm={{action this.sendMessage "info"}}
+/>
+```
+
+Within `ButtonWithConfirmation`, the code in the `submitConfirm` action does not
+change. It will still invoke `onConfirm` without explicit arguments:
+
+```javascript {data-filename=app/components/button-with-confirmation.js}
+const promise = this.onConfirm();
+```
+
+However the expression `{{action this.sendMessage "info"}}` used in passing the
+action to the component creates a closure, i.e. an object that binds the
+parameter we've provided to the function specified. So now when the action is
+invoked, that parameter will automatically be passed as its argument,
+effectively calling `sendMessage("info")`, despite the argument not appearing in
+the calling code.
+
+So far in our example, the action we have passed to `ButtonWithConfirmation` is
+a function that accepts one argument, `messageType`. Suppose we want to extend
+this by allowing `sendMessage` to take a second argument, the actual text of the
+message the user is sending:
+
+```javascript {data-filename=app/components/send-message.js}
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class SendMessage extends Component {
+  @action
+  sendMessage(messageType, messageText) {
+    // send message here and return a promise
+  }
+}
+```
+
+We want to arrange for the action to be invoked from within
+`ButtonWithConfirmation` with both arguments. We've seen already that if we
+provide a `messageType` value to the `action` helper when we insert
+`ButtonWithConfirmation` into its parent `SendMessage` template, that value will
+be passed to the `sendMessage` action as its first argument automatically when
+invoked as `onConfirm`. If we subsequently pass a single additional argument to
+`onConfirm` explicitly, that argument will be passed to `sendMessage` as its
+second argument (This ability to provide arguments to a function one at a time
+is known as [currying](https://en.wikipedia.org/wiki/Currying)).
+
+In our case, the explicit argument that we pass to `onConfirm` will be the
+required `messageText`. However, remember that internally our
+`ButtonWithConfirmation` component does not know or care that it is being used
+in a messaging application. Therefore within the component's JavaScript file, we
+will use a property `confirmValue` to represent that argument and pass it to
+`onConfirm` as shown here:
+
+```javascript {data-filename=app/components/button-with-confirmation.js}
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class ButtonWithConfirmation extends Component {
+  @action
+  submitConfirm() {
+    if (this.args.onConfirm) {
+      // call `onConfirm` with a second argument
+      await this.args.onConfirm(this.confirmValue);
+    }
+
+    this.showConfirmation = false;
+  }
+
+  //...
+}
+```
+
+In order for `confirmValue` to take on the value of the message text, we'll bind
```

Comment:
I'm a little concerned about too much overloading of the term "bind". It could mean (and is used in the guides to refer to each of the following)

* data-binding
* creation of a new variable binding (i.e., `{{#let apple as |grape|}}` or `const grape = apple`)
* `Function.prototype.bind`

To avoid confusion, we may want to take care to differentiate very clearly